src/project/ProjectModel.js

```diff
@@ -59,6 +59,13 @@ define(function (require, exports, module) {
     var _exclusionListRegEx = /\.pyc$|^\.git$|^\.gitmodules$|^\.svn$|^\.DS_Store$|^Thumbs\.db$|^\.hg$|^CVS$|^\.hgtags$|^\.idea$|^\.c9revisions$|^\.SyncArchive$|^\.SyncID$|^\.SyncIgnore$|\~$/;
 
     /**
+     * @public
+     * Glob definition of files and folders that should be excluded directly
+     * inside node domain watching with chokidar
+     */
+    var exclusionGlob = "**/(.pyc|.git|.gitmodules|.svn|.DS_Store|Thumbs.db|.hg|CVS|.hgtags|.idea|.c9revisions|.SyncArchive|.SyncID|.SyncIgnore|node_modules)";
```

Comment:
done in https://github.com/adobe/brackets/pull/12647/commits/147b5eb84a894678809bd385e849ac183f339cac
ad `.git`, extension can call (and calls)
```
FileSystem.prototype.watch = function (entry, filter, filterGlobs, callback) {
```
with any arguments (filters) it wants, there's no reason to watch insides of `.git` when no git extension is installed